control-base/src/main/java/fi/nls/oskari/control/feature/AbstractWFSFeaturesHandler.java

```diff
@@ -127,19 +133,17 @@ protected SimpleFeatureCollection getFeatures(String id, String uuid, OskariLaye
         // TODO: Figure out the maxFeatures from the layer
         int maxFeatures = 10000;
 
-        Filter filter = null;
-        if (myPlacesHelper.isMyPlacesLayer(layer)) {
-            int categoryId = myPlacesHelper.getCategoryId(id);
-            filter = myPlacesHelper.getFilter(categoryId, uuid, bbox);
-        } else if (userlayerHelper.isUserlayerLayer(layer)) {
-            int userlayerId = userlayerHelper.getUserlayerId(id);
-            filter = userlayerHelper.getFilter(userlayerId, uuid, bbox);
-        }
+        // Find out if we need custom filter or postProcessing for features
+        Optional<UserLayerService> processor =
+                userContentProcessors.stream()
+                .filter(proc -> proc.isUserContentLayer(id))
+                .findAny();
+        Filter filter = processor.map(proc-> proc.getWFSFilter(id, uuid, bbox)).orElse(null);
```

Comment:
Got bored of chaining method calls so the map and orElse calls are separate from the original chain? :)